memphis88/week4/ass5/app/components/assignment-app.jsx

```diff
@@ -0,0 +1,71 @@
+var React = require('react');
+
+var OnPressingEnterMixin = require('../mixins/on-pressing-enter-mixin');
+var GeneratedList = require('./generated-list');
+var ListActions = require('../actions/list-actions');
+var ListStore = require('../stores/list-store');
+
+var AssignmentApp = React.createClass({
+	mixins: [
+		OnPressingEnterMixin
+	],
+
+	propTypes: {
+		generatedLists: React.PropTypes.array,
+		listName: React.PropTypes.string,
+	},
+
+	getInitialState: function() {
+		return {
+			generatedLists: ListStore.getAll(),
+			listName: ''
```

Comment:
You could put a [ref](https://facebook.github.io/react/docs/more-about-refs.html) on your input field instead. Get its value by `this.refs.inputField.value;` Clear it out with `this.refs.inputField.value = "";` Then you don't have to maintain state and make the view more stateless.